mkosi/qemu.py

```diff
@@ -646,20 +646,62 @@ def finalize_firmware_variables(
     stack: contextlib.ExitStack,
 ) -> tuple[Path, str]:
     ovmf_vars = stack.enter_context(tempfile.NamedTemporaryFile(prefix="mkosi-ovmf-vars-"))
+    ovmf_json = stack.enter_context(tempfile.NamedTemporaryFile(prefix="mkosi-ovmf-vars-json-"))
+    have_mok_json = False
+
+    if config.secure_boot_certificate:
+        try:
+            from virt.firmware.efi import guids, siglist
```

Comment:
Is there a sensible way to get this within the Python stdlib? Right now we don't have external dependencies. `guids.parse_str` only seems to generate a UUID object, which is from the stdlib, but `mok_sigdb.add_cert` seems to read the raw bytes out of the X509 cert, which is a tad more difficult.

Reply:
Shelling out to `virt-fw-vars` is much better than importing the module. Shelling out means we can use `virt-fw-vars` from a tools tree if there is one, whereas with importing we can't do that without going through extra hoops. Please rework this to use the `virt-fw-vars` command line tool, looking at its `--help` there seems to be options to do what you're doing here.